### PR TITLE
added support for providing validity for cert.

### PR DESCRIPTION
**Pull request by vlahane**
## Description

If you haven't already, read the full [contribution guide](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md). The guide may have changed since the last time you read it, so please double-check. Once you are done and ready to submit your PR, edit the PR description and run through the relevant checklist below.

Enable GitHub Worksflows in your fork to enable auto-generation of assets with our [GitHub Action](/.github/workflows/push-auto-generate.yml).
To trigger GitHub Actions after auto-generation, [add a GitHub PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) as a secret in your forked repository called `PAT`.

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [ ] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [x] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.
- [ ] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [x] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [ ] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [ ] I have updated the examples in README with the latest module version number.

### CODE REVIEW

**Comment by vlahane** [comment]
locally brm validate working fine but failed in pipeline, anyone knows the actual cause? 

`The file "/home/runner/work/bicep-registry-modules/bicep-registry-modules/modules/deployment-scripts/create-kv-certificate/main.json" is modified or outdated. Please regenerate the file to fix it.`

**Comment by vlahane** [comment]
@dciborow  do you have any idea ?

**Comment by dciborow** [comment]
> @dciborow do you have any idea ?

looks like you may need to run "brm generate". You could also try setting up the GitHub action in your fork that runs the auto-gen for you.
```
Enable GitHub Worksflows in your fork to enable auto-generation of assets with our [GitHub Action](https://github.com/.github/workflows/push-auto-generate.yml).
To trigger GitHub Actions after auto-generation, [add a GitHub PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) as a secret in your forked repository called PAT.
```

I didn't realize you had put up this PR until after I gave it a try. But I think I like the way you did it better.

**Comment by vlahane** [comment]
> > @dciborow do you have any idea ?
> 
> looks like you may need to run "brm generate". You could also try setting up the GitHub action in your fork that runs the auto-gen for you.
> 
> Enable GitHub Worksflows in your fork to enable auto-generation of assets with our [GitHub Action](https://github.com/.github/workflows/push-auto-generate.yml). To trigger GitHub Actions after auto-generation, [add a GitHub PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) as a secret in your forked repository called PAT.
> 
> I didn't realize you had put up this PR until after I gave it a try. But I think I like the way you did it better.

I did tried `brm generate` and `brm validate` locally works fine but In PR it's failing

File: modules/deployment-scripts/create-kv-certificate/main.bicep [hunk 1]
```diff
@@ -58,6 +58,9 @@ param organizationId string = ''
 @description('Override this parameter if using this in cross tenant scenarios')
 param isCrossTenant bool = false
 
+@description('Optional. Override default validityInMonths 12 value')
```
Comment (dciborow) [inline modules/deployment-scripts/create-kv-certificate/main.bicep:61]:
```suggestion
@minValue(1)
@maxValue(1200)
@description('Optional. Override default validityInMonths 12 value')
```

File: modules/deployment-scripts/create-kv-certificate/create-kv-cert.sh [hunk 2]
```diff
@@ -30,6 +31,7 @@ do
       fi
       policy=$(az keyvault certificate get-default-policy \
         | sed -e s/CN=CLIGetDefaultPolicy/CN=${certCommonName}/g  \
+        | sed -e s/\"validityInMonths\":\ 12/\"validityInMonths\":\ ${validity}/g \
```
Comment (dciborow) [inline modules/deployment-scripts/create-kv-certificate/create-kv-cert.sh:34]:
```suggestion
        | sed -e s/\"validityInMonths\":\ 12/\"validityInMonths\":\ ${validity}/g \
        | sed -e s/CN=CLIGetDefaultPolicy/CN=${certCommonName}/g  \
```

Comment (dciborow) [reply modules/deployment-scripts/create-kv-certificate/create-kv-cert.sh:34]: lets set the values in the same order in the both places .

**Comment by dciborow** [comment]
> > > @dciborow do you have any idea ?
> > 
> > 
> > looks like you may need to run "brm generate". You could also try setting up the GitHub action in your fork that runs the auto-gen for you.
> > Enable GitHub Worksflows in your fork to enable auto-generation of assets with our [GitHub Action](https://github.com/.github/workflows/push-auto-generate.yml). To trigger GitHub Actions after auto-generation, [add a GitHub PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) as a secret in your forked repository called PAT.
> > I didn't realize you had put up this PR until after I gave it a try. But I think I like the way you did it better.
> 
> I did tried `brm generate` and `brm validate` locally works fine but In PR it's failing

I left a few suggestions. Maybe after applying them, you could try to run 'brm generate' one more time. Changing the contents of the bicep file should really force a regen. Also just double check that when you are adding the files with git that you are adding main.json?

If I still see it stuck, Ill pull your branch onto my machine and take a look.

**Comment by dciborow** [comment]
> > > @dciborow do you have any idea ?
> > 
> > 
> > looks like you may need to run "brm generate". You could also try setting up the GitHub action in your fork that runs the auto-gen for you.
> > Enable GitHub Worksflows in your fork to enable auto-generation of assets with our [GitHub Action](https://github.com/.github/workflows/push-auto-generate.yml). To trigger GitHub Actions after auto-generation, [add a GitHub PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) as a secret in your forked repository called PAT.
> > I didn't realize you had put up this PR until after I gave it a try. But I think I like the way you did it better.
> 
> I did tried `brm generate` and `brm validate` locally works fine but In PR it's failing

O this looks good!
image

I think that's going to take care of it for you.

**Review by dciborow — APPROVED** [review]

**Comment by vlahane** [comment]
@dciborow thank you!

**Comment by dciborow** [comment]
Everything looks good, just have to wait on https://github.com/Azure/bicep-registry-modules/issues/286 now.

**Comment by dciborow** [comment]
> @dciborow thank you!

oops, one more thing, we need to update the version in version.json to 3.1, and the versions in the readme to 3.1.1, you can take a peak here.
https://github.com/Azure/bicep-registry-modules/pull/304

**Comment by vlahane** [comment]
> > @dciborow thank you!
> 
> oops, one more thing, we need to update the version in version.json to 3.1, and the versions in the readme to 3.1.1, you can take a peak here. #304

updated!

**Comment by dciborow** [comment]
idk why the validate-module-files error is back. Touching the readme shouldn't have changed anything. Making some improvements over here though based on what im seeing. https://github.com/Azure/bicep-registry-modules/pull/305

**Comment by dciborow** [comment]
@vlahane , https://github.com/vlahane/bicep-registry-modules/pull/1

**Comment by vlahane** [comment]
> @vlahane , [vlahane#1](https://github.com/vlahane/bicep-registry-modules/pull/1)

what was this cause here?

**Comment by dciborow** [comment]
> > @vlahane , [vlahane#1](https://github.com/vlahane/bicep-registry-modules/pull/1)
> 
> what was this cause here?

I think in this commit something weird happened to the main.json
image

I'm going to adjust the triggers on the autofix action though so that it just fixes something like this if any of the files are out of sync.

**Comment by vlahane** [comment]
> > > @vlahane , [vlahane#1](https://github.com/vlahane/bicep-registry-modules/pull/1)
> > 
> > 
> > what was this cause here?
> 
> I think in this commit something weird happened to the main.json image
> 
> I'm going to adjust the triggers on the autofix action though so that it just fixes something like this if any of the files are out of sync.

I think in git action, it should replace main.json even if user ran or not ran brm command  or main.json updated in commit ?

**Review by shenglol — APPROVED** [review]
:shipit:

**Pull request merged**